Clarify deregistration content (#1374)
sensu/sensu-docs: content/sensu-core/1.7/reference/clients.md

Similarly to registration events, Sensu can create and process a deregistration event when the Sensu client process stops.
You can use deregistration events to trigger a handler that updates external CMDBs or performs an action to update ephemeral infrastructures.

> By default, registration events trigger the built-in deregistration extension. To enable deregistration events, set the client `deregister` flag to `true`. This will result in clients being immediately depopulated from the dashboard. If you require an external servide to be handle the deregistration event (and specify the event handler using the client [`deregistration` definition scope][48].

Reviewer: ```suggestion
By default, registration events trigger the built-in deregistration extension. To enable deregistration events, set the client `deregister` flag to `true`. This will result in clients being immediately depopulated from the dashboard. If you require an external service to be handle the deregistration event (and specify the event handler using the client [`deregistration` definition scope][48].
```

Reviewer: Looks like there's also an errant `(` here.

> By default, registration events trigger the built-in deregistration extension. To enable deregistration events, set the client `deregister` flag to `true`. This will result in clients being immediately depopulated from the dashboard. If you require an external service to be handle the deregistration event (and specify the event handler using the client [`deregistration` definition scope][48].

Reviewer: ```suggestion
By default, deregistration events trigger the built-in deregistration extension. To enable deregistration events, set the client `deregister` flag to `true`. This results in the client being immediately depopulated from the dashboard in the event that the client process stops. If you require an external service to handle the deregistration event, specify the deregistration event handler using the client [`deregistration` definition scope][48].
```

> By default, deregistration events trigger the built-in deregistration extension. To enable deregistration events, set the client `deregister` flag to `true`. This results in the client being immediately depopulated from the dashboard in the event that the client process stops. If you require an external service to handle the deregistration event, specify the deregistration event handler using the client [`deregistration` definition scope][48].

Reviewer 2: ```suggestion
By default, deregistration events trigger the built-in deregistration extension. To enable deregistration events, set the client `deregister` flag to `true`. This results in the client being immediately depopulated from the client registry in the event that the sensu-client process is administratively stopped. If you require additional logic or actions to be applied when deregistration events are received, specify your custom deregistration event handler using the client [`deregistration` definition scope][48].
```

> _NOTE: When setting `"deregister": true`, the default behavior is for the respective client to be immediately depopulated from the dashboard. We recommend instead that users implement a handler inside of the "keepalive" scope, as this will deregister clients based on a keepalive exceeding a threshold. If you require that an external service be notified about the deregistration, we recommend using the  Chef, Puppet, or AWS/EC2 plugins for your respective version (Core, Enterprise) of Sensu._

Reviewer: ```suggestion
_NOTE: When the client `deregister` flag is set to `true`, the default behavior is for the respective client to be immediately depopulated from the dashboard in the event that the client process stops. However, to ensure visibility into stopped clients, we recommend implementing a handler inside the keepalive scope to deregister clients based on a keepalive exceeding a threshold. If you require that an external service be notified about client deregistration, we recommend using the Chef, Puppet, or AWS/EC2 plugins for your respective version (Core, Enterprise) of Sensu._
```

> _NOTE: When the client `deregister` flag is set to `true`, the default behavior is for the respective client to be immediately depopulated from the dashboard in the event that the client process stops. However, to ensure visibility into stopped clients, we recommend implementing a handler inside the keepalive scope to deregister clients based on a keepalive exceeding a threshold. If you require that an external service be notified about client deregistration, we recommend using the Chef, Puppet, or AWS/EC2 plugins for your respective version (Core, Enterprise) of Sensu._

Reviewer 2: I think we need more words outside this note to describe the differences between deregistration and keepalives. Trying to fit it all into this note isn't the best way to give the complete picture.
